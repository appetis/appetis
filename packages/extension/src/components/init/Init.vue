<template>
  <div
    class="fixed inset-0 overflow-y-auto z-10"
    aria-labelledby="modal-title"
    role="dialog"
    aria-modal="true"
  >
    <div class="init-modal-background" aria-hidden="true"></div>
    <div class="modal">
      <!--      <span class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true">&#8203;</span>-->

      <div class="modal-content-xl">

        <div class="flex">
          <Left />
          <div class="w-6/12 relative">
            <Header title="Are you new?" />
            <div class="modal-body">
              <h2 class="text-lg pt-4 pb-8">Do you have an invite code?</h2>

              <div class="text-right flex">
                <div class="w-1/2 px-1">
                  <button
                    class="btn-border-black"
                    @click="$emit('goSignup')"
                    id="btn-no"
                  >
                    No
                  </button>
                </div>
                <div class="w-1/2 px-1">
                  <button
                    class="btn-border-black"
                    @click="$emit('goHasCode')"
                    id="btn-yes"
                  >
                    Yes
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Left from '@/components/init/Left'
import Header from '@/components/init/Header'
export default {
  name: 'Init',
  props: {},
  components: {
    Left,
    Header
  },
  data() {
    return {}
  },
}
</script>
