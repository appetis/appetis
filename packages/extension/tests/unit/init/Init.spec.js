import Init from '@/components/init/Init'
import { mount } from '@vue/test-utils'

const factory = () => {
  const wrapper = mount(Init)
  return wrapper
}

describe('init/Init', () => {
  it('init modal snapshot', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("doesn't have an invite code", () => {
    const wrapper = factory()
    const mockGoSignup = jest.fn()
    wrapper.vm.$on('goSignup', mockGoSignup)
    wrapper.find('#btn-no').trigger('click')

    expect(mockGoSignup).toHaveBeenCalledTimes(1)
  })

  it('has an invite code', () => {
    const wrapper = factory()
    const mockGoCode = jest.fn()
    wrapper.vm.$on('goHasCode', mockGoCode)
    wrapper.find('#btn-yes').trigger('click')

    expect(mockGoCode).toHaveBeenCalledTimes(1)
  })
})
