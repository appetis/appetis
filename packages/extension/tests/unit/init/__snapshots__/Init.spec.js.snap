// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init/Init init modal snapshot 1`] = `
<div aria-labelledby="modal-title" role="dialog" aria-modal="true" class="fixed inset-0 overflow-y-auto z-10">
  <div aria-hidden="true" class="init-modal-background"></div>
  <div class="modal">
    <div class="modal-content-sm">
      <div class="modal-body">
        <h2 class="text-lg pt-4 pb-8">Do you have an invite code?</h2>
        <div class="text-right flex">
          <div class="w-1/2 px-1"><button id="btn-no" class="btn-border-black">
              No
            </button></div>
          <div class="w-1/2 px-1"><button id="btn-yes" class="btn-border-black">
              Yes
            </button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
